lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers_test.go

```diff
@@ -1044,6 +1044,108 @@ func TestGetSubscriber(t *testing.T) {
 	tests.RunUnitTest(t, e, tc)
 }
 
+func TestGetSubscriberByExactIMSI(t *testing.T) {
+	// Regression test to ensure we are loading states with the exact same IMSI
+	// key as the subscriber
+
+	configuratorTestInit.StartTestService(t)
+	deviceTestInit.StartTestService(t)
+	stateTestInit.StartTestService(t)
+	err := configurator.CreateNetwork(configurator.Network{ID: "n1"}, serdes.Network)
+	assert.NoError(t, err)
+
+	e := echo.New()
+	getSubscriberURL := "/magma/v1/lte/:network_id/subscribers/:subscriber_id"
+	createSubscribersURL := "/magma/v1/lte/:network_id/subscribers_v2"
+	handlers := handlers.GetHandlers()
+	getSubscriber := tests.GetHandlerByPathAndMethod(t, handlers, getSubscriberURL, obsidian.GET).HandlerFunc
+	createSubscribers := tests.GetHandlerByPathAndMethod(t, handlers, createSubscribersURL, obsidian.POST).HandlerFunc
+
+	// Pre: create two APNs
+	_, err = configurator.CreateEntities(
+		"n1",
+		[]configurator.NetworkEntity{
+			{Type: lte.APNEntityType, Key: "apn0"},
+			{Type: lte.APNEntityType, Key: "apn1"},
+		},
+		serdes.Entity,
+	)
+	assert.NoError(t, err)
+
+	// Create two subscribers, one a prefix of the other
+	sub1 := newMutableSubscriber("IMSI1234567890")
+	sub2 := newMutableSubscriber("IMSI123456789000")
+	sub2.Name = "John Doe"
+	payload := subscriberModels.MutableSubscribers{sub1, sub2}
+
+	tc := tests.Test{
+		Method:         "POST",
+		URL:            createSubscribersURL,
+		Payload:        tests.JSONMarshaler(payload),
+		Handler:        createSubscribers,
+		ParamNames:     []string{"network_id"},
+		ParamValues:    []string{"n1"},
+		ExpectedStatus: 201,
+	}
+	tests.RunUnitTest(t, e, tc)
+
+	tc = tests.Test{
+		Method:         "GET",
+		URL:            getSubscriberURL,
+		Handler:        getSubscriber,
+		ParamNames:     []string{"network_id", "subscriber_id"},
+		ParamValues:    []string{"n1", "IMSI1234567890"},
+		ExpectedStatus: 200,
+		ExpectedResult: sub1.ToSubscriber(),
+	}
+	tests.RunUnitTest(t, e, tc)
+
+	// Now create some AGW-reported state for the subscribers
+	// First we need to register a gateway which can report state
+	_, err = configurator.CreateEntity(
+		"n1",
+		configurator.NetworkEntity{Type: orc8r.MagmadGatewayType, Key: "g1", Config: &models.MagmadGatewayConfigs{}, PhysicalID: "hw1"},
+		serdes.Entity,
+	)
+	assert.NoError(t, err)
```

Comment:
Use this convenience fn instead https://github.com/magma/magma/blob/dc72fd4842888768f0e2974b964d99d873cc4f23/orc8r/cloud/go/services/configurator/test_utils/utils.go#L33-L35